auth_saml_create_user/models/res_users.py

```diff
@@ -0,0 +1,35 @@
+# © 2019 Savoir-faire Linux
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+import logging
+
+from odoo import api, models
+
+_logger = logging.getLogger(__name__)
+
+
+class ResUsers(models.Model):
+    _inherit = "res.users"
+
+    @api.multi
+    def _auth_saml_signin(self, provider, validation, saml_response):
+        saml_uid = validation['user_id']
+        user_ids = self.search(
+            [('saml_uid', '=', saml_uid), ('saml_provider_id', '=', provider)])
+        if self.check_if_create_user(provider) and not user_ids:
+            self.create_user(saml_uid, provider)
+        return super()._auth_saml_signin(provider, validation, saml_response)
+
+    def check_if_create_user(self, provider):
+        return self.env['auth.saml.provider'].browse(provider).create_user
+
+    def create_user(self, saml_uid, provider):
+        _logger.debug("Creating new Odoo user \"%s\" from SAML" % saml_uid)
+        SudoUser = self.env['res.users'].sudo()
+        new_user = SudoUser.create({
+            'name': saml_uid,
+            'login': saml_uid,
+            'saml_provider_id': provider,
+            'company_id': self.env['res.company'].sudo().browse(1).id,
```

Comment:
browse(1)? better to not use id directly, ids could be changed

Reply:
In fact, the feature that would extend and improve this module is to allow the creation of 'template users'.Right now, the "My Company will be used". So I will create an issue for this as a feature request.

Reply:
```suggestion
            'company_id': self.env.ref('base.main_company').id
```